Tests/EventListener/IsGrantedListenerTest.php

```diff
@@ -82,18 +100,20 @@ public function testAccessDeniedMessages(array $attributes, $subject, $expectedM
             ->method('isGranted')
             ->will($this->returnValue(false));
 
-        $listener = new IsGrantedListener($authChecker);
-        $isGranted = new IsGranted(array('attributes' => $attributes, 'subject' => $subject));
-        $request = $this->createRequest($isGranted);
-
         // avoid the error of the subject not being found in the request attributes
+        $arguments = array();
         if (null !== $subject) {
-            $request->attributes->set($subject, 'bar');
+            $arguments[$subject] = 'bar';
         }
 
-        $this->setExpectedException(AccessDeniedException::class, $expectedMessage);
+        $listener = new IsGrantedListener($this->createArgumentNameConverter($arguments), $authChecker);
+        $isGranted = new IsGranted(array('attributes' => $attributes, 'subject' => $subject));
+        $request = $this->createRequest($isGranted);
+
+        $this->expectException(AccessDeniedException::class);
```

Comment:
you need to keep support for PHPUnit 4.8 here, as PHP 5.5 is not supported by newer versions.